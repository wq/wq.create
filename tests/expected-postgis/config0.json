{
    "pages": {
        "login": {
            "url": "login",
            "name": "login"
        },
        "logout": {
            "url": "logout",
            "name": "logout"
        },
        "index": {
            "url": "",
            "icon": "directions",
            "verbose_name": "Map",
            "description": "Project overview map",
            "section": "Contributions",
            "order": 0,
            "map": {
                "mapId": "map",
                "layers": [
                    {
                        "name": "Observations",
                        "type": "geojson",
                        "url": "/observations.geojson",
                        "popup": "observation"
                    }
                ]
            },
            "name": "index"
        },
        "category": {
            "icon": "config",
            "description": "Manage available categories",
            "section": "Admin",
            "order": 100,
            "show_in_index": "can_change",
            "cache": "all",
            "background_sync": false,
            "name": "category",
            "url": "categories",
            "list": true,
            "form": [
                {
                    "name": "name",
                    "label": "Name",
                    "bind": {
                        "required": true
                    },
                    "wq:length": 255,
                    "type": "string"
                },
                {
                    "name": "description",
                    "label": "Description",
                    "type": "text"
                }
            ],
            "verbose_name": "category",
            "verbose_name_plural": "categories",
            "label_template": "{{name}}"
        },
        "observation": {
            "icon": "list",
            "description": "View and submit photos on map",
            "section": "Contributions",
            "order": 1,
            "cache": "first_page",
            "background_sync": true,
            "map": [
                {
                    "mode": "list",
                    "layers": [
                        {
                            "name": "Observations",
                            "type": "geojson",
                            "data": [
                                "context_feature_collection",
                                "geometry"
                            ],
                            "popup": "observation",
                            "cluster": true
                        },
                        {
                            "name": "All Observations",
                            "type": "vector-tile",
                            "layer": "observation",
                            "popup": "observation",
                            "identifyLayers": [
                                "observation-circle"
                            ],
                            "active": false
                        }
                    ]
                },
                {
                    "mode": "detail",
                    "layers": [
                        {
                            "name": "Observation",
                            "type": "geojson",
                            "data": [
                                "context_feature",
                                "geometry"
                            ],
                            "popup": "observation"
                        },
                        {
                            "name": "All Observations",
                            "type": "vector-tile",
                            "layer": "observation",
                            "popup": "observation",
                            "identifyLayers": [
                                "observation-circle"
                            ],
                            "active": false
                        }
                    ]
                },
                {
                    "mode": "edit",
                    "layers": []
                }
            ],
            "name": "observation",
            "url": "observations",
            "list": true,
            "form": [
                {
                    "name": "date",
                    "label": "Date",
                    "hint": "The date when the observation was taken",
                    "type": "date"
                },
                {
                    "name": "category",
                    "label": "Category",
                    "hint": "Observation type",
                    "type": "select one",
                    "wq:ForeignKey": "category",
                    "wq:related_name": "observation_set"
                },
                {
                    "name": "geometry",
                    "label": "Location",
                    "bind": {
                        "required": true
                    },
                    "hint": "The location of the observation",
                    "type": "geopoint"
                },
                {
                    "name": "photo",
                    "label": "Photo",
                    "hint": "Photo of the observation",
                    "type": "image"
                },
                {
                    "name": "notes",
                    "label": "Notes",
                    "hint": "Field observations and notes",
                    "type": "text",
                    "multiline": true
                }
            ],
            "verbose_name": "observation",
            "verbose_name_plural": "observations",
            "ordering": [
                "-date"
            ],
            "label_template": "{{date}}",
            "geometry_fields": [
                {
                    "name": "geometry",
                    "label": "Location",
                    "type": "geopoint"
                }
            ]
        }
    },
    "site_title": "test Project",
    "router": {
        "base_url": ""
    },
    "store": {
        "service": "",
        "defaults": {
            "format": "json"
        }
    },
    "logo": "/static/app/images/icon-192.png",
    "material": {
        "theme": {
            "primary": "#7500ae",
            "secondary": "#0088bd"
        }
    },
    "map": {
        "bounds": [
            [
                -180,
                -70
            ],
            [
                180,
                70
            ]
        ],
        "tiles": "/tiles/{z}/{x}/{y}.pbf"
    },
    "debug": true
}
