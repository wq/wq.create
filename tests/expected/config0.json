{
    "pages": {
        "login": {
            "url": "login",
            "name": "login"
        },
        "logout": {
            "url": "logout",
            "name": "logout"
        },
        "category": {
            "icon": "config",
            "description": "Manage available categories",
            "section": "Admin",
            "show_in_index": "can_change",
            "cache": "all",
            "background_sync": false,
            "name": "category",
            "url": "categories",
            "list": true,
            "form": [
                {
                    "name": "name",
                    "label": "Name",
                    "bind": {
                        "required": true
                    },
                    "wq:length": 255,
                    "type": "string"
                },
                {
                    "name": "description",
                    "label": "Description",
                    "type": "text"
                }
            ],
            "verbose_name": "category",
            "verbose_name_plural": "categories",
            "label_template": "{{name}}"
        },
        "observation": {
            "icon": "list",
            "description": "View and submit photos on map",
            "section": "Contributions",
            "cache": "first_page",
            "background_sync": true,
            "map": [
                {
                    "mode": "list",
                    "autoLayers": true,
                    "layers": []
                },
                {
                    "mode": "detail",
                    "autoLayers": true,
                    "layers": []
                },
                {
                    "mode": "edit",
                    "layers": []
                }
            ],
            "name": "observation",
            "url": "observations",
            "list": true,
            "form": [
                {
                    "name": "date",
                    "label": "Date",
                    "hint": "The date when the observation was taken",
                    "type": "date"
                },
                {
                    "name": "category",
                    "label": "Category",
                    "hint": "Observation type",
                    "type": "select one",
                    "wq:ForeignKey": "category",
                    "wq:related_name": "observation_set"
                },
                {
                    "name": "geometry",
                    "label": "Location",
                    "bind": {
                        "required": true
                    },
                    "hint": "The location of the observation",
                    "type": "geopoint"
                },
                {
                    "name": "photo",
                    "label": "Photo",
                    "hint": "Photo of the observation",
                    "type": "image"
                },
                {
                    "name": "notes",
                    "label": "Notes",
                    "hint": "Field observations and notes",
                    "type": "text",
                    "multiline": true
                }
            ],
            "verbose_name": "observation",
            "verbose_name_plural": "observations",
            "ordering": [
                "-date"
            ],
            "label_template": "{{date}}"
        },
        "index": {
            "url": "",
            "name": "index",
            "show_in_index": false,
            "verbose_name": "test Project"
        }
    },
    "site_title": "test Project",
    "logo": "/icon-192.png",
    "router": {
        "base_url": ""
    },
    "store": {
        "service": "",
        "defaults": {
            "format": "json"
        }
    },
    "material": {
        "theme": {
            "primary": "#7500ae",
            "secondary": "#0088bd"
        }
    },
    "map": {
        "bounds": [
            [
                -180,
                -70
            ],
            [
                180,
                70
            ]
        ]
    },
    "debug": true
}
